lib/BeeswaxClient.js

```diff
@@ -236,4 +237,103 @@ BeeswaxClient.prototype._delete = function(endpoint, idField, id, failOnNotFound
     });
 };
 
+BeeswaxClient.prototype.uploadCreativeAsset = function(params){
+    var self = this;
+
+    function init(params){
+       var assetDef = {}, 
+           props = ['advertiser_id','creative_asset_name','size_in_bytes','notes','active'];
+       
+        props.forEach(function(prop){
+            if (params[prop]){
+                assetDef[prop] = params[prop];
+            }
+        });
+
+        if (!params.sourceUrl) {
+            return Promise.reject(
+                new Error('uploadCreativeAsset params requires a sourceUrl property.')
+            );
+        }
+
+
+        if (!assetDef.creative_asset_name){
+            assetDef.creative_asset_name = 
+                urlUtils.parse(params.sourceUrl).pathname.split('/').pop();
+        }
+       
+        return Promise.resolve({ req : params, assetDef : assetDef });
+    }
+
+    function getSize(data) {
+        if (data.assetDef.size_in_bytes && data.assetDef.size_in_bytes > 0){
+            return Promise.resolve(data);
+        }
+
+        return self.request('head', { url : data.req.sourceUrl }).then(function(body) {
```

Comment:
You should probably just use plain old `request-promise` here instead of the Beeswax client's `request` method, as the client's request method does extra stuff like attaching the cookie jar to the options and attempting to authenticate to beeswax on 401 errors.